CMakeLists.txt

```diff
@@ -2,174 +2,229 @@
+    ## Setup Trilinos as TPL
+    ##---------------------------------------------------------------------------##
 
-ASSERT_DEFINED(${PACKAGE_NAME}_ENABLE_EXPLICIT_INSTANTIATION)
+    SET(${PACKAGE_NAME}_TRILINOS_TPL ON)
 
-TRIBITS_ADD_OPTION_AND_DEFINE(
-    ${PACKAGE_NAME}_ENABLE_EXPLICIT_INSTANTIATION
-    HAVE_DATATRANSFERKIT_EXPLICIT_INSTANTIATION
-    "Enable explicit template instantiation (ETI) in DTK"
-    ${${PACKAGE_NAME}_ENABLE_EXPLICIT_INSTANTIATION}
-    )
+    ##---------------------------------------------------------------------------##
+    ## FLAGS
+    ##---------------------------------------------------------------------------##
 
-INCLUDE("${CMAKE_CURRENT_SOURCE_DIR}/cmake/ETI.cmake")
+    SET(${PACKAGE_NAME}_ENABLE_CXX11_DEFAULT ON)
 
-##---------------------------------------------------------------------------##
-## Set the path to the DTKData directory.
-##---------------------------------------------------------------------------##
+    ##---------------------------------------------------------------------------##
+    ## TriBITS package processing
+    ##---------------------------------------------------------------------------##
 
-# Search for one of the mesh files in the DTKData directory
-find_file(DTK_DATA_MESH_FILE cube_mesh.exo PATHS
-    ${${PACKAGE_NAME}_SOURCE_DIR}/data/exodus
-    ${${PACKAGE_NAME}_SOURCE_DIR}/DTKData/exodus
-    NO_DEFAULT_PATH
-)
-if(DTK_DATA_MESH_FILE)
-    # Get the location of the mesh file
-    get_filename_component(DTK_DATA_DIR ${DTK_DATA_MESH_FILE} DIRECTORY)
-    # Strip "/exodus" from the path
-    string(REGEX REPLACE "/exodus" "" DTK_DATA_DIR ${DTK_DATA_DIR})
-    message("-- DTKData found at ${DTK_DATA_DIR}")
-else()
-    set(DTK_DATA_DIR DTK_DATA_DIR-NOTFOUND)
-    message("-- DTKData NOT found")
-endif()
+    # This will launch the second parsing of this file in ELSE() branch
+    TRIBITS_PROJECT()
 
-##---------------------------------------------------------------------------##
-## Print the revision number to stdout
-##---------------------------------------------------------------------------##
-
-FIND_PACKAGE(Git)
-IF(GIT_FOUND AND EXISTS ${${PACKAGE_NAME}_SOURCE_DIR}/.git)
-    EXECUTE_PROCESS(
-        COMMAND           ${GIT_EXECUTABLE} log --pretty=format:%H -n 1
-        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-        OUTPUT_VARIABLE   ${PACKAGE_NAME}_GIT_COMMIT_HASH
-        )
 ELSE()
-    SET(${PACKAGE_NAME}_GIT_COMMIT_HASH "Not a git repository")
-ENDIF()
-MESSAGE("DTK Revision = '${${PACKAGE_NAME}_GIT_COMMIT_HASH}'")
+    INCLUDE(CombinedOption)
+
+    TRIBITS_PACKAGE(${PACKAGE_NAME})
+    TRIBITS_ADD_DEBUG_OPTION()
+    TRIBITS_ADD_SHOW_DEPRECATED_WARNINGS_OPTION()
+
+    ##---------------------------------------------------------------------------##
+    ## ETI types
+    ##---------------------------------------------------------------------------##
+
+    IF(${PACKAGE_NAME}_TRILINOS_TPL)
```

Comment:
Why do we have an IF ELSE here?

Reply:
(I think) 
If we run Trilinos as TPL, this is inside the second pass, where we need to grab device types and ETI configs from Trilinos. Otherwise, if DTK is being built as part of Trilinos, we can simply use the cached Tpetra vars.